_posts/2023-01-10-cleaning-up-the-trino-backlog.md

```diff
@@ -0,0 +1,109 @@
+if we'd automatically closed those!
+
+The result of the effort so far has been excellent.
+
+<img src="/assets/blog/backlog-blog/open-pull-requests-graph.png"/>
+
+We're not completely done with the cleanup effort, but as you can see, we're
+slowing down. Our oldest PRs are increasingly recent, still in development,
+and worth having open. Going from a peak of 700+ open pull requests to around
+300 is a massive improvement, and the goal is to end up in the vicinity of about
+200 open pull requests in Trino at any point in time.
+
+But with the cleanup being so manual, the next challenge is stopping the pull
+requests from steadily piling back up while we're not paying attention to them.
+The fix for that is simple - we're going to keep paying attention. The Trino
+developer relations team is planning on tracking and getting involved in two
+categories of pull requests to keep the number of open PRs stable.
+
+The first category is pull requests that don't get any immediate attention from
+a reviewer. While Trino reviewers are overall excellent and quick to take a look
+at incoming pull requests, about five percent slip through the cracks, where a
+contributor submits something that receives no reviews or comments and lives on
+in the pull request backlog. That's not a good experience for the contributor,
+and it's not good for Trino, either, because that contribution could have a lot
+of value. We plan on stopping this from happening by implementing workflows
+which spring Trino developer relations into action when these situations arise.
+If a pull request goes a few days without a comment, we'll be the safety net to
+ask questions, get engineers involved, and make sure that at least a few pairs
+of eyes take a look at every incoming PR in a timely manner.
+
+The second category is pull requests that get some reviews, but eventually
+stagnate or stop being actively worked on. This happens for a lot of reasons,
+but in all cases, if a pull request goes a few weeks with no activity, the
+developer relations team will be checking in. Our goal will be to figure out the
+proper path forward, whether that's flagging down some reviewers again,
+communicating that the pull request should be closed, or anything else. The end
+result should be that nothing slips through the cracks and ends up going months
+without human contact. If an author vanishes or everyone gets too busy to look
+at a pull request again, though, the final stop will ultimately be a stale bot
+which closes pull requests that have gone a few months with no activity.
+
+With all these processes in place, contributors should never feel like their
+efforts are going unnoticed. Submitted code should be reviewed quickly,
+iterated on in a timely manner, and merged without much delay. In situations
+where a pull request is *not* going to be merged, the Trino developer relations
+should be able to chime in quickly to make that clear, saving contributors from
```

Comment:
```suggestion
where a pull request is *not* going to be merged, the Trino developer relations
team should be able to chime in quickly to make that clear, saving contributors from
```

Reply:
Just commenting this so I don't forget, I'll tackle it when addressing other feedback.